rasa/nlu/classifiers/diet_classifier.py

```diff
@@ -1102,47 +1131,76 @@ def _combine_sparse_dense_features(
 
         return tf.concat(dense_features, axis=-1) * mask
 
+    def _features_as_seq_ids(
+        self, features: List[Union[np.ndarray, tf.Tensor, tf.SparseTensor]], name: Text,
+    ) -> tf.Tensor:
+        # if there are dense features it's enough
+        for f in features:
+            if not isinstance(f, tf.SparseTensor):
+                return tf.stop_gradient(f)
+
+        # we need dense labels for negative sampling
+        for f in features:
```

Comment:
I think you can combine the two for loops as you return in both cases.
```
        for f in features:
            if not isinstance(f, tf.SparseTensor):
                return tf.stop_gradient(f)
            else:
                return tf.stop_gradient(
                    self._tf_layers[f"sparse_to_dense_ids.{name}"](f)
                )
```

Reply:
what if they're in different order? I want to avoid unnecessary multiplication if the second feature is dense

Reply:
Ah, sorry. You could do something like:
```
dense_features_present = any([not isinstance(f, tf.SparseTensor) for f in features])

if dense_features_present:
...
```

Reply:
would it be the same number of forms effectively?

Reply:
because I'll need to find these features in the next for

Reply:
forms?